src/main/java/org/prebid/server/bidder/conversant/ConversantBidder.java

```diff
@@ -130,97 +129,105 @@ private static App updateApp(App app, String siteId) {
     }
 
     private static List<Integer> makeApi(List<Integer> extApi, List<Integer> videoApi) {
-        final List<Integer> protocols = CollectionUtils.isNotEmpty(extApi) ? extApi : videoApi;
-        return CollectionUtils.isNotEmpty(protocols)
-                ? protocols.stream().filter(APIS::contains).collect(Collectors.toList()) : videoApi;
+        final List<Integer> api = CollectionUtils.isNotEmpty(extApi) ? extApi : videoApi;
+        return CollectionUtils.isNotEmpty(api)
+                ? api.stream().filter(APIS::contains).collect(Collectors.toList())
+                : videoApi;
     }
 
     private static List<Integer> makeProtocols(List<Integer> extProtocols, List<Integer> videoProtocols) {
         final List<Integer> protocols = CollectionUtils.isNotEmpty(extProtocols) ? extProtocols : videoProtocols;
         return CollectionUtils.isNotEmpty(protocols)
-                ? protocols.stream().filter(PROTOCOLS::contains).collect(Collectors.toList()) : videoProtocols;
+                ? protocols.stream().filter(PROTOCOLS::contains).collect(Collectors.toList())
+                : videoProtocols;
     }
 
     @Override
     public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
         try {
-            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
-            return Result.withValues(extractBids(httpCall.getRequest().getPayload(), bidResponse));
-        } catch (DecodeException | PreBidException e) {
+            return Result.withValues(extractBids(httpCall));
+        } catch (PreBidException e) {
             return Result.withError(BidderError.badServerResponse(e.getMessage()));
         }
     }
 
-    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+    private List<BidderBid> extractBids(HttpCall<BidRequest> httpCall) {
+        final BidResponse bidResponse;
+        try {
+            bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+        } catch (DecodeException e) {
+            throw new PreBidException(String.format("bad server response: %s. ", e.getMessage()));
+        }
         if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
             throw new PreBidException("Empty bid request");
         }
-        return bidsFromResponse(bidRequest, bidResponse);
+        return bidsFromResponse(httpCall.getRequest().getPayload(), bidResponse);
     }
 
     private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
         final SeatBid firstSeatBid = bidResponse.getSeatbid().get(0);
-        return firstSeatBid.getBid().stream()
+        final List<Bid> bids = firstSeatBid.getBid();
+
+        if (CollectionUtils.isEmpty(bids)) {
+            throw new PreBidException("Empty bids array");
+        }
+        return bids.stream()
```

Comment:
didn't see that check

Reply:
Rethinking...
Seems you're right, but this check is very similar to existing one - `Empty bid request`.
So, what we should do if `seatbid.bid` array is empty?
i'd say it was for compatibility. What do you think?